src/App.js

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+import { Matrix } from "./components/MatrixBlock";
+import { MatrixBuilder } from "./components/MatrixBuilder";
+import styles from "./index.scss";
+
+function App() {
+  const [data, setData] = useState(Array());
+  const [settings, setSettings] = useState({
+    rows: 0,
+    columns: 0,
+    cells: 0,
+  });
+
+  const handleSetMatrix = (data) => setData(data);
+
+  return (
+    <div className={styles.app}>
+      <MatrixBuilder
+        data={data}
+        handleSetMatrix={handleSetMatrix}
+        settings={settings}
+        onSetSettings={setSettings}
+      />
+      <Matrix
+        data={data}
+        settings={settings}
+        handleSetMatrix={handleSetMatrix}
```

Comment:
no need setSettings in a Matrix component

Reply:
In Matrix we have a function that push row in our array, and we use handleSetMatrix for setting this in state

Reply:
no need. just set the previous matrix set with a new row. we should handle matrix data not a settings